### PR TITLE
[openvpn] Ignore UNDEF nodes #59

### PR DESCRIPTION
**Pull request by EdgeKing810**
GCI Task: https://codein.withgoogle.com/dashboard/task-instances/4806628416159744/

I'm trying to make a test to see if I can get those ``UNDEF`` nodes for now...

### CODE REVIEW

File: tests/test_openvpn.py [hunk 1]
```diff
@@ -8,6 +8,7 @@
 
 CURRENT_DIR = os.path.dirname(os.path.realpath(__file__))
 links2 = open('{0}/static/openvpn-2-links.txt'.format(CURRENT_DIR)).read()
+links2u = open('{0}/static/openvpn-2-links-undef.txt'.format(CURRENT_DIR)).read()
```
Comment (nemesisdesign) [inline tests/test_openvpn.py]: rename this var to `links2undef` which improves readability

File: tests/test_openvpn.py [hunk 2]
```diff
@@ -19,6 +20,12 @@ def test_parse(self):
         self.assertEqual(p.version, '1')
         self.assertEqual(p.metric, 'static')
 
+    def test_parse_undef(self):
+        p = OpenvpnParser(links2u)
+        self.assertIsInstance(p.graph, networkx.Graph)
+        self.assertEqual(p.version, '1')
+        self.assertEqual(p.metric, 'static')
```
Comment (nemesisdesign) [inline tests/test_openvpn.py]: I think here we should add:
```
self.assertEqual(len(data['nodes']), 0)
self.assertEqual(len(data['links']), 0)
```
That's what I'd expect, do these assertions pass?

File: netdiff/parsers/openvpn.py [hunk 3]
```diff
@@ -42,6 +42,8 @@ def parse(self, data):
             links = data.routing_table.values()
         # add clients in graph as nodes
         for client in clients:
+            if (client.common_name == "UNDEF"):
```
Comment (nemesisdesign) [inline netdiff/parsers/openvpn.py]: round parentheses not needed and use single quotes for consistency please:

```python
if client.common_name == 'UNDEF':
```

**Comment by EdgeKing810** [comment]
If travis was a real guy, I'd probably hate him tbh. :|

**Pull request closed without merging**